### PR TITLE
Auth and home

### PR DESCRIPTION
**Pull request by Shailseen**
**Files to be reviewed:**
- redux
  - features
    - authSlice.js
    - postSlice.js
- routes
   - Home
   - Login
- component
  - CardIcon
  - Loader
  - Modal
  - ModalPostCard
  - Navbar
  - NewPostCard
  - PostCard

[Preview Link](https://deploy-preview-1--moments-chat.netlify.app/)

### CODE REVIEW

File: src/feature/authSlice.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+const authSlice = createSlice({
+  name: "auth",
+  initialState,
+  reducers: {
+    logout: (state) => {
+      state.status = STATUSES.IDLE;
+      state.user = {};
+      localStorage.removeItem("token");
+    //   toast('Logged out successfully!')
```
Comment (aniketsingh13) [inline src/feature/authSlice.js]: remove the commented code

File: src/feature/authSlice.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      .addCase(signupUser.fulfilled, (state, action) => {
+        console.log(action.payload);
+        state.status = STATUSES.IDLE;
+        // state.user = {
+        //   token: action.payload.encodedToken,
+        //   firstName: action.payload.createdUser.firstName,
+        //   lastName: action.payload.createdUser.lastName,
+        // };
+        // localStorage.setItem("token", action.payload.encodedToken);
+        // toast(`Welcome ${state.user.firstName + state.user.lastName}`);
```
Comment (aniketsingh13) [inline src/feature/authSlice.js:54]: here also

File: src/feature/authSlice.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      const { username, password } = data;
+      const res = await axios.post("/api/auth/login", {
+        username: username,
+        password: password,
+      });
+      console.log(res)
+      return res.data;
+    } catch (error) {
+        console.log(error)
+    //   return thunkAPI.rejectWithValue(error);
```
Comment (aniketsingh13) [inline src/feature/authSlice.js]: here

File: src/index.js [hunk 4]
```diff
@@ -3,13 +3,25 @@ import ReactDOM from "react-dom";
...
-    <App />
+    <BrowserRouter>
+      <Provider store={store}>
+        <Routes>
+          <Route path="/" element={<App />}>
+            <Route path="/" element={<Login />} />
+          </Route>
+        </Routes>
+      </Provider>
+    </BrowserRouter>
```
Comment (aniketsingh13) [inline src/index.js:26]: can make seperate folder for routes

File: src/components/Loader/Loader.module.css [hunk 5]
```diff
@@ -0,0 +1,30 @@
+.loader {
+  width: 48px;
+  height: 48px;
```
Comment (aniketsingh13) [inline src/components/Loader/Loader.module.css:3]: use rem instead of px

File: src/components/Loader/Loader.module.css [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+}
+.loader::after {
+  content: "";
+  box-sizing: border-box;
+  position: absolute;
+  left: 50%;
+  top: 50%;
+  transform: translate(-50%, -50%);
+  width: 44px;
+  height: 44px;
```
Comment (aniketsingh13) [inline src/components/Loader/Loader.module.css:19]: here also

**Review by aniketsingh13 — APPROVED** [review]
ui look good 
code is easy to read
please remove commented code

**Review by Rohitprasad83 — APPROVED** [review]
- Code is readable
- Good naming conventions used

File: src/index.js (hunk 4 above)
Comment (Shailseen) [reply src/index.js:26]: yes , i will do this later

**Pull request merged**